frb_codegen/src/ir/ty_type_impl_trait.rs

```diff
@@ -0,0 +1,86 @@
+use crate::ir::*;
+use crate::target::Target;
+use convert_case::{Case, Casing};
+use syn::Ident;
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct IrTypeImplTrait {
+    pub trait_bounds: Vec<String>,
+}
+
+impl IrTypeImplTrait {
+    pub fn join(&self) -> String {
```

Comment:
uber-nit: not very sure whether it is a good idea to have "IrTypeImplTrait.join", which does not seem very clear

Reply:
I remove it, just use `to_enum`.